pallets/gear/src/manager/task.rs

```diff
@@ -154,4 +154,9 @@ where
     fn wake_message(&mut self, _program_id: ProgramId, _message_id: MessageId) {
         todo!("issue #349");
     }
+
+    fn remove_gas_reservation(&mut self, reservation_id: ReservationId) {
+        log::debug!("Remove reservation with {:?}", reservation_id);
+        Pallet::<T>::consume_and_retrieve(reservation_id);
```

Comment:
You must update here reservation's map of the program or otherwise you will have non-sync state of the gas and program storages.

There are 2 ways the gas reservation can be removed: 1) by the call to `unreserve` in the current execution, or 2) by the task processor (on reservation deadline). The first way updates reservations on `unreserve` so in the next program execution no removed reservations occur, while the second way removes reservation only from the gas tree, but leaves it in the program's reservations map. Next time when program tries to unreserve it, the sys-call won't fail, but we will reach panic because of the try to consume non-existing reservation.